benchmark/benchmark_N.py

```diff
@@ -0,0 +1,53 @@
+import time
+import math
+import socket
+from pathlib import Path
+
+import numpy as np
+import tensorflow as tf
+import matplotlib.pyplot as plt
+
+import utils
+import displace
+
+# gpu = tf.config.experimental.list_physical_devices('GPU')[0]
+# tf.config.experimental.set_memory_growth(gpu, True)
+# tf.debugging.set_log_device_placement(True)
+
+total_t = []
+loop_t = []
+mat_sizes = [400, 800, 1600, 3200]
+num_alphas = [100, 1000, 5000]
+for N in mat_sizes:
+    b_size = math.floor(2.5e8 / N ** 2)
+    loop_it = []
+    loop_lt = []
+    for num in num_alphas:
+        alphas = utils.random_alphas(num)
+        alphas = tf.data.Dataset.from_tensor_slices(alphas).batch(b_size)
+
+        start_time = time.perf_counter()
+        f = displace.gen_displace_BCH(N)
+        init_time = time.perf_counter() - start_time
+
+        for batch in alphas:
+            f(batch)
+        loop_time = time.perf_counter() - init_time - start_time
+
+        loop_it.append(init_time + loop_time)
+        loop_lt.append(loop_time)
+        print("%i\t%i\t%.3f\t%.3f" % (N, num, init_time + loop_time, loop_time))
+    total_t.append(loop_it)
+    loop_t.append(loop_lt)
+
+fig, ax = plt.subplots(1, 1)
+ax.set_ylabel("Time (s)")
+ax.set_xlabel("Number of Displacements")
+# ax.set_yscale("log")
+plt.grid(True, which="both")
+for idx, N in enumerate(mat_sizes):
+    ax.plot(num_alphas, total_t[idx], label="Total-{}".format(N))
```

Comment:
Change the label to something like `label="Total-N={}".format(N)` , I was confused about what these numbers meant in the plot.